### PR TITLE
Fixed misleading error message for `rasa train nlu`

### LINKED ISSUE #5587: invalid file specified via `--nlu` gives misleading error message
**Rasa version**: 1.9.5

**Python version**: 3.7

**Operating system** (windows, osx, ...): macOS

**Issue**:

If the file passed to `rasa train nlu --nlu nlu_data.md` is not in the right format, Rasa returns a misleading error message, `No NLU data given. Please provide NLU data in order to train a Rasa NLU model using the '--nlu' argument.` It should give an error about the format of the file instead.

**Error (including full traceback)**:
```
No NLU data given. Please provide NLU data in order to train a Rasa NLU model using the '--nlu' argument.
```

**Command or request that led to error**:
```
rasa train nlu --nlu nlu_data.md
```

**Content of nlu_data.md file** :
```yml
## deny
- non, merci
- non merci
- non

## affirm
- oui
- oui, merci
- oui merci
```

### PR DESCRIPTION
**Pull request by degiz**
Closes #5587 

**Proposed changes**:

Clarify a situation when the command is given a file with a wrong format

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: data/test/markdown_single_sections/incorrect_nlu_format.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+## deny
```
Comment (wochinge) [inline data/test/markdown_single_sections/incorrect_nlu_format.md:1]: optional / up to you: Inline this to make the test more concise (it avoid having to jump to the fixture and then the file and then back to the test) and avoid us having a ton of files in the `data/test/` directories.

File: tests/test_train.py [hunk 2]
```diff
@@ -129,3 +130,22 @@ def test_train_nlu_temp_files(
...
+    monkeypatch.setattr(tempfile, "tempdir", tmp_path)
+
+    train_nlu(
+        default_stack_config,
+        incorrect_nlu_data,
+        output="test_train_nlu_temp_files_models",
+    )
+
+    captured = capsys.readouterr()
+    assert "Please verify the data format" in captured.out
```
Comment (wochinge) [inline tests/test_train.py:151]: should we also test that the actual training is not running?
And is there a test for the `No NLU data given.` case?

File: rasa/train.py [hunk 3]
```diff
@@ -431,10 +431,16 @@ def train_nlu(
 
     training_datas = await file_importer.get_nlu_data()
     if training_datas.is_empty():
-        print_error(
-            "No NLU data given. Please provide NLU data in order to train "
-            "a Rasa NLU model using the '--nlu' argument."
-        )
+        if nlu_data:
+            print_error(
+                f"Path '{nlu_data}' doesn't contain valid NLU data in it. "
```
Comment (wochinge) [inline rasa/train.py:436]: I think we should also explicitly state that the training of the NLU model will be skipped.

File: changelog/5587.misc.rst [hunk 4]
```diff
@@ -0,0 +1 @@
+Fixed misleading error message for a situation when ``rasa train nlu --nlu`` is given a file with a wrong format.
```
Comment (wochinge) [inline changelog/5587.misc.rst:1]: should we mention the old and new error message?

File: data/test/markdown_single_sections/incorrect_nlu_format.md (hunk 1 above)
Comment (degiz) [reply data/test/markdown_single_sections/incorrect_nlu_format.md:1]: I thought about that in the first place as well.
The thing is this test is testing the "train" command e2e, so it actually has to read a real file and open() it.

File: tests/test_train.py (hunk 2 above)
Comment (degiz) [reply tests/test_train.py:151]: that's a very good point! Yes, the situation with `No NLU data given` can still happen imo. 
I've added one more tests case!

**Pull request merged**